cmd/bosun/cache/cache.go

```diff
@@ -0,0 +1,48 @@
+package cache // import "bosun.org/cmd/bosun/cache"
+
+import (
+	"bosun.org/_third_party/github.com/golang/groupcache/lru"
+	"bosun.org/_third_party/github.com/golang/groupcache/singleflight"
+	"sync"
+)
+
+type Cache struct {
+	g singleflight.Group
+
+	sync.Mutex
+	lru *lru.Cache
+}
+
+// obj is an LRU object tracking data and a corresponding error.
+type obj struct {
+	Val interface{}
+	Err error
+}
+
+func New(MaxEntries int) *Cache {
+	return &Cache{
+		lru: lru.New(MaxEntries),
+	}
+}
+
+func (c *Cache) Get(key string, getFn func() (interface{}, error)) (interface{}, error) {
+	// for simplicity we use a global lock on the LRU
+	// this might cause races where an object appears right after finding it doesn't exist
```

Comment:
This line and the one below are no longer true since we changed the locking in the get fn.